kivy/ext/mpl/backend_kivy.py

```diff
@@ -0,0 +1,832 @@
+from kivy.graphics import Rectangle
+from kivy.uix.widget import Widget
+from kivy.uix.label import Label
+from kivy.uix.floatlayout import FloatLayout
+from kivy.uix.behaviors import FocusBehavior
+from kivy.uix.actionbar import ActionBar, ActionView, \
+                                ActionButton, ActionToggleButton, \
+                                ActionPrevious, ActionOverflow, ActionSeparator
+from kivy.base import EventLoop
+from kivy.core.text import Label as CoreLabel
+from kivy.graphics import Color, Line
+from kivy.graphics import Rotate, Translate
+from kivy.graphics.tesselator import Tesselator
+from kivy.graphics.context_instructions import PopMatrix, PushMatrix
+from kivy.logger import Logger
+from kivy.graphics import Mesh
+from kivy.resources import resource_find
+from kivy.uix.stencilview import StencilView
+from kivy.core.window import Window
+from kivy.uix.button import Button
+from kivy.uix.boxlayout import BoxLayout
+from kivy.uix.floatlayout import FloatLayout
+from kivy.uix.relativelayout import RelativeLayout
+from kivy.uix.popup import Popup
+from kivy.properties import ObjectProperty
+from kivy.clock import Clock
+
+import numpy as np
+import io
+from functools import partial
+from math import cos, sin, pi
+
+# try:
+#     kivy.require('1.9.0') # I would need to check which release of
+#     Kivy would be the best suitable.
+# except AttributeError:
+#     raise ImportError(
+#         "kivy version too old -- it must have require_version")
+
+_debug = True
+
+app = None
+
+
+class SaveDialog(FloatLayout):
+    save = ObjectProperty(None)
+    text_input = ObjectProperty(None)
+    cancel = ObjectProperty(None)
+
+
+class MPLKivyApp(App):
+
+    figure = ObjectProperty(None)
+    toolbar = ObjectProperty(None)
+
+    def build(self):
+        EventLoop.ensure_window()
+        layout = FloatLayout()
+        if self.figure is not None:
+            self.figure.size_hint_y = 0.9
+            layout.add_widget(self.figure)
+        if self.toolbar is not None:
+            self.toolbar.size_hint_y = 0.1
+            layout.add_widget(self.toolbar)
+        return layout
+
+    def on_stop(self):
+        global app
+        app = None
+
+
+def _create_App(fig_canvas, toolbar):
+    global app
+    if app is None:
+        if _debug:
+            print("Starting up Kivy Application")
+        app = MPLKivyApp(figure=fig_canvas, toolbar=toolbar)
+
+
+class RendererKivy(RendererBase):
+    """
+    The renderer handles drawing/rendering operations.
+
+    This is a minimal do-nothing class that can be used to get started when
+    writing a new backend. Refer to backend_bases.RendererBase for
+    documentation of the classes methods.
+    """
+    def __init__(self, dpi, widget):
+        self.dpi = dpi
+        self.widget = widget
+        #  Can be enhanced by using TextToPath matplotlib, textpath.py
+        self.mathtext_parser = MathTextParser("Bitmap")
+        self.list_goraud_triangles = []
```

Comment:
I would prefer that any instance variables you create are declared first at class level so please init these in the class with e.g. `_clipd = None`.